lms/djangoapps/commerce/views.py

```diff
@@ -0,0 +1,179 @@
+    permission_classes = (IsAuthenticated,)
+
+    def _is_data_valid(self, request):
+        """
+        Validates the data posted to the view.
+
+        Arguments
+            request -- HTTP request
+
+        Returns
+            Tuple (data_is_valid, course_key, error_msg)
+        """
+        data = request.DATA or {}
+        course_id = data.get('course_id', None)
+
+        if not course_id:
+            # Translators: Do not translate the term course_id.
+            return False, None, _('course_id field is missing.')
+
+        try:
+            course_key = CourseKey.from_string(course_id)
+            courses.get_course(course_key)
+        except Exception as ex:  # pylint: disable=broad-except
+            log.exception('Unable to locate course matching %s.', course_id)
+            return False, None, ex.message
+
+        return True, course_key, None
+
+    def _get_jwt(self, user):
+        """
+        Returns a JWT object with the specified user's info.
+        """
+        data = {
+            'username': user.username,
+            'email': user.email
+        }
+        return jwt.encode(data, getattr(settings, 'ECOMMERCE_API_SIGNING_KEY'))
+
+    def _enroll(self, course_key, user):
+        """ Enroll the user in the course. """
+        CourseEnrollment.enroll(user, course_key, True)
+
+    def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Attempt to purchase the course seat and enroll the user.
+        """
+        user = request.user
+        valid, course_key, error = self._is_data_valid(request)
+        if not valid:
+            return DetailResponse(error, status=HTTP_406_NOT_ACCEPTABLE)
+
+        # Ensure that the E-Commerce API is setup properly
+        ecommerce_api_url = getattr(settings, 'ECOMMERCE_API_URL')
+        ecommerce_api_signing_key = getattr(settings, 'ECOMMERCE_API_SIGNING_KEY')
+
+        if not (ecommerce_api_url and ecommerce_api_signing_key):
+            self._enroll(course_key, user)
+            msg = 'E-Commerce API not setup. Enrolled {} in {} directly.'.format(user.username, unicode(course_key))
+            log.debug(msg)
+            return DetailResponse(msg)
+
+        # Default to honor mode. In the future we may expand this view to support additional modes.
+        mode = CourseMode.DEFAULT_MODE_SLUG
+        course_modes = CourseMode.objects.filter(course_id=course_key, mode_slug=mode, sku__isnull=False)
+
+        # If there are no course modes with SKUs, enroll the user without contacting the external API.
+        if not course_modes.exists():
+            msg = _('The %(enrollment_mode)s mode for %(course_id)s does not have a SKU. '
+                    'Enrolling %(username)s directly.')
+            msg = msg % {
+                'enrollment_mode': mode,
+                'course_id': unicode(course_key),
+                'username': user.username
+            }
```

Comment:
I have had issues with `ugettext` and `.format()` in the past. I cannot recall the specifics; but, I'd like to avoid the possibility of that being an issue in the future.